src/brainglobe_napari/napari_atlas_representation.py

```diff
@@ -0,0 +1,27 @@
+from dataclasses import dataclass
+
+from bg_atlasapi import BrainGlobeAtlas
+from napari.viewer import Viewer
+
+
+@dataclass
+class NapariAtlasRepresentation:
+    """Representation of a BG atlas as napari layers"""
+
+    bg_atlas: BrainGlobeAtlas
+
+    def add_to_viewer(self, viewer: Viewer):
```

Comment:
My intial thought here was maybe it would make more sense to have a function that looks like:
```python
def add_to_viewer(bg_atlas: BrainGlobeAtlas, viewer: Viewer) -> NapariAtlasRepresentation:
    ...
```
And have `NapariAtlasRepresentation` store references to the viewer layers?

I think that would depend on what the intended use of `NapariAtlasRepresentation` is after creation though - perhaps including a short tutorial/example of using `NapariAtlasRepresentation` in this PR would help clarify that?

Reply:
Good point - I'm not sure about the exact intended use.

I went down this route because it was simpler to relinquish responsibility for napari objects (e.g. other plugins can do what they want without depending on BG-specific data structures)? I didn't want to deal with the case that another plugin deletes/changes a napari layer that we store a reference to :joy: 

What advantages does the suggested design bring?

An advantage of the current design is that the `NapariAtlasRepresentation` can easily query stuff about the atlas it represents.

So maybe the question is whether we want `NapariAtlasRepresentation` to be more tightly coupled to `napari` or to the BG Atlas API? (and feeds into a wider question

(Either way, good idea to add an example in this PR)

Reply:
My query is what's the point of a whole `NapariAtlasRepresentation` class (that doesn't have any additional attributes/methods) over just a funciton to add an atlas to a viewer? The only state `NapariAtlasRepresentation` stores is to a `BrainglobeAtlas`, so it would be easier to just interact with the `BrainglobeAtlas` directly. Perhaps there's context or plans for future development I'm missing?

Reply:
Right, I see your point!

TLDR: I can definitely see the `NapariAtlasRepresentation` class becoming at least a bit more complicated in the future.
At some point the way atlases are represented in `napari` will change. It might be good for the widget code to not have to care about this?

Longer version:

The `NapariAtlasRepresentation` class is _likely_ to have some (probably private) functions that check some stuff about the atlas (does it have meshes (not all atlases do)? is it symmetric?) and then respond to this (if it has meshes, add some mesh layers to the viewer)? These will be called inside `add_to_viewer`... 

Also, at some point we'd like to represent the atlas as nested layers (when napari allows, see figure in #1 )

It's _possible_ that we might want to have the widget store references to layers in the future (and e.g. a "remove atlas button", or `create_atlas_from_layers()` or `get_atlas_info_from_layer` or...) but I wanted to avoid this complexity for now.

It's also possible that users tell us they want separate buttons to add the annotation, the reference and the meshes, in which case `add_to_viewer` would become three functions `add_annotation_to_viewer()`, `add_reference_to_viewer`, ...

It might also be good to not reimplement any functionality the `BrainGlobeAtlas` class provides or could provide, so I was thinking this class should purely know about how to relay info from `BrainGlobeAtlas` to `napari`.

Maybe I am being too OO about this :joy: 

@adamltyson do you have more thoughts about what users might want to do with atlases in napari in the future?

@dstansby if I didn't manage to convince you with this, I'll happily take your suggested approach!

Reply:
@alessandrofelder sorry just seen this. Not sure exactly what users might need to do, but I think your idea is the right one. 

> this class should purely know about how to relay info from BrainGlobeAtlas to napari.

This is what I was thinking.